LandSandBoat/server
# [dbtool] Add function to dump DB to SQL files

**_I affirm:_**

## What does this pull request do?

Adds a function to the dbtool maintenance tasks section to dump database data in-place into the sql files.

Queries database then scans corresponding SQL file, replacing values in INSERT statements with data from the database.

Is not smart about which INSERT statement gets which data; line-end comments can get shuffled around if SQL file INSERTs are out of order or missing. **If you're doing INSERT or DELETE operations, make sure to add a blank ``INSERT INTO `table` VALUES ();`` or manually remove the INSERT statement respectively before dumping tables.**

Almost everything except dbtool and item_latents is just formatting changes from the tool. item_latents has had some manual reordering and comment unshuffling done.

There's some issues with Python correctly padding floats according to the database settings, so trailing 0s are trimmed all throughout. None of the values should have actually changed, and the padding is readded when imported into the database. I'm not sure if there's an easy way around this, but it's just a visual thing.

## Steps to test these changes


1. Import a fresh database.
2. Perform a query on your DB to update some set of values.
3. Run dbool, select Maintenance Tasks (t), dump either a single table or the whole db (d or a), check git diffs and verify only your changed values have changed.
    - Alternatively use `py -3 dbtool.py dump` or `py -3 dbtool.py dump TABLE_NAME`.
